analysis/study_definition.py

```diff
@@ -1,33 +1,191 @@
+
+def first_diagnosis_in_period(dx_codelist):
+    return patients.with_these_clinical_events(
+        dx_codelist,
+        returning="date",
+        find_first_match_in_period=True,
+        include_month=True,
+        return_expectations={
+            "incidence": 0.2,
+            "date": {"earliest": "1950-01-01", "latest": "today"},
+        },
+    )
+
+def medication_count_3m_0m(med_codelist):
+    return patients.with_these_medications(
+        med_codelist,
+        between=["2019-12-01", "2020-02-29"],
+        returning="number_of_matches_in_period",
+        return_expectations={
+            "int": {"distribution": "normal", "mean": 3, "stddev": 2},
+            "incidence": 0.1,
+        },
+    )
+
+def medication_count_6m_3m(med_codelist):
+    return patients.with_these_medications(
+        med_codelist,
+        between=["2019-09-01", "2020-11-30"],
+        returning="number_of_matches_in_period",
+        return_expectations={
+            "int": {"distribution": "normal", "mean": 3, "stddev": 2},
+            "incidence": 0.1,
+        },
+    )
+
+def medication_count_12m_6m(med_codelist):
+    return patients.with_these_medications(
+        med_codelist,
+        between=["2019-03-01", "2020-08-31"],
+        returning="number_of_matches_in_period",
+        return_expectations={
+            "int": {"distribution": "normal", "mean": 3, "stddev": 2},
+            "incidence": 0.1,
+        },
+    )
+
+def medication_earliest(med_codelist):
+    return patients.with_these_medications(
+        med_codelist,
+        between=["2010-01-01", "2020-02-29"],
+        returning="date",
+        find_first_match_in_period=True,
+        include_month=True,
+        return_expectations={
+            "incidence": 0.2,
+            "date": {"earliest": "2010-01-01", "latest": "2020-02-29"},
+        },
+    )
+
+def medication_latest(med_codelist):
+    return patients.with_these_medications(
+        med_codelist,
+        between=["2010-01-01", "2020-02-29"],
+        returning="date",
+        find_last_match_in_period=True,
+        include_month=True,
+        return_expectations={
+            "incidence": 0.2,
+            "date": {"earliest": "2010-01-01", "latest": "2020-02-29"},
+        },
+    )
+
+def medication_counts_and_dates(var_name, med_codelist_file):
+    """
+    Generates dictionary of covariats for a medication including counts and dates
+    
+    Takes a variable prefix and medication codelist filename (minus .csv)
+    Returns a dictionary suitable for unpacking into the main study definition
+    This will include all five of the items defined in the functions above
+    """
+    
+    definitions={}
+    if (med_codelist_file[0:5] == "cross"):
+        med_codelist_file = "crossimid-codelists/" + med_codelist_file
+    else:
+        med_codelist_file = "codelists/" + med_codelist_file
+    med_codelist=codelist_from_csv(med_codelist_file + ".csv", system="snomed", column="snomed_id")
```

Comment:
this is good idea BUT I'm pretty sure we aren't 100% consistent in the meds codelists and there might be further changes with a shift to new BNF coding system.

I should check with team but I think best thing might be for you to run and see hwere it breaks and we fix column names codelist by codelist  on OpenCodelists

Reply:
Ok. The codelists are consistent for now since most of them were generated from the same source.